common/lib/xmodule/xmodule/js/src/word_cloud/word_cloud_main.js

```diff
@@ -154,136 +158,188 @@
+                );
             }
 
             $.each(response.student_words, function(word, stat) {
                 var percent = (response.display_student_percents) ? ' ' + (Math.round(100 * (stat / response.total_count))) + '%' : '';
 
-                studentWordsKeys.push('<strong>' + word + '</strong>' + percent);
+                studentWordsKeys.push(HtmlUtils.interpolateHtml(
+                    '{listStart}{startTag}{word}{endTag}{percent}{listEnd}',
+                    {
+                        listStart: HtmlUtils.HTML('<li>'),
+                        startTag: HtmlUtils.HTML('<strong>'),
+                        word: word,
+                        endTag: HtmlUtils.HTML('</strong>'),
+                        percent: percent,
+                        listEnd: HtmlUtils.HTML('</li>')
+                    }
+                ).toString());
             });
-            studentWordsStr = '' + studentWordsKeys.join(', ');
+
+            studentWordsStr = '' + studentWordsKeys.join('');
 
             cloudSectionEl
-            .addClass('active')
-            .find('.your_words').html(studentWordsStr)
-            .end()
-            .find('.total_num_words').html(response.total_count);
+                .addClass('active');
+
+            HtmlUtils.setHtml(
+                cloudSectionEl.find('.your_words'),
+                HtmlUtils.HTML(studentWordsStr)
+            );
+
+            HtmlUtils.setHtml(
+                cloudSectionEl.find('.your_words').end().find('.total_num_words'),
+                HtmlUtils.interpolateHtml(
+                    gettext('{start_strong}{total}{end_strong} words submitted in total.'),
+                    {
+                        start_strong: HtmlUtils.HTML('<strong>'),
+                        end_strong: HtmlUtils.HTML('</strong>'),
+                        total: response.total_count
+                    }
+                )
+            );
 
             $(cloudSectionEl.attr('id') + ' .word_cloud').empty();
 
-        // Actual drawing of word cloud.
+            // Actual drawing of word cloud.
             groupEl = d3.select('#' + cloudSectionEl.attr('id') + ' .word_cloud').append('svg')
-                .attr('width', this.width)
-                .attr('height', this.height)
-                .append('g')
-                .attr('transform', 'translate(' + (0.5 * this.width) + ',' + (0.5 * this.height) + ')')
-                .selectAll('text')
-                .data(words)
-                .enter().append('g');
+                    .attr('width', this.width)
+                    .attr('height', this.height)
+                    .append('g')
+                    .attr('transform', 'translate(' + (0.5 * this.width) + ',' + (0.5 * this.height) + ')')
+                    .selectAll('text')
+                    .data(words)
+                    .enter()
+                    .append('g')
+                    .attr('data-id', function() {
```

Comment:
This seems overly complex. Why can't you just loop over the words and use a simple counter (0, 1, 2, 3...)?